File: c/arr.c
#include <stdio.h>
#include <string.h>

void copy_array(int src[], int dst[], int size)
{
    memcpy(dst, src, size);
}

void print_array(int arr[], int size)
{
    printf("[");
    for (int i = 0; i < size; i++)
    {
        if (i < size - 1)
        {
            printf("%d ", arr[i]);
        }
        else
        {
            printf("%d", arr[i]);
        }
    }
    printf("]");
}

File: go/arr/arr.go
package arr

import "fmt"

func CopyArray(src []int, dst []int, size int) {
	for i := 0; i < size; i++ {
		dst[i] = src[i]
	}
}

func PrintArray(arr []int, size int) {
	fmt.Print("[")
	for i := 0; i < size; i++ {
		if i < size-1 {
			fmt.Printf("%d ", arr[i])
		} else {
			fmt.Printf("%d", arr[i])
		}
	}
	fmt.Print("]")
}
